### PR TITLE
Updated platform docs

### PR DESCRIPTION
**Pull request by obulat**
I updated platform documentation to include Winforms instead of Win32 as a backend for Windows. I also added Android as a backend. I'm not sure about the status of development there.

## PR Checklist:

- [ ] All new features have been tested
- [ ] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

File: docs/reference/platforms.rst [hunk 1]
```diff
@@ -13,22 +13,22 @@ OS X
...
+The backend for OS X is named `toga_cocoa`_. It supports OS X 10.7 (Lion)
 and later. It is installed automatically on OS X machines (machines that
 report ``sys.platform == 'darwin'``), or can be manually installed by invoking::
 
     $ pip install toga[cocoa]
 
 The OS X backend has seen the most development to date.
 
-.. _toga-cocoa: http://github.com/pybee/toga-cocoa
+.. _toga-cocoa: https://github.com/pybee/toga/tree/master/src/cocoa
```
Comment (goanpeca) [inline docs/reference/platforms.rst]: Should this be 
`.. _toga_cocoa:`

File: docs/reference/platforms.rst [hunk 2]
```diff
@@ -38,29 +38,30 @@ invoking::
...
+.. _toga-winforms: https://github.com/pybee/toga/tree/master/src/winforms
 
 Mobile platforms
 ----------------
 
 iOS
 ~~~
 
-The backend for iOS is named `toga-iOS`_. It supports iOS 6 or later. It
+The backend for iOS is named `toga_iOS`_. It supports iOS 6 or later. It
```
Comment (goanpeca) [inline docs/reference/platforms.rst]: Also here, why changing the `toga_iOS` to `toga-iOS`?

File: docs/reference/platforms.rst (hunk 1 above)
Comment (obulat) [reply docs/reference/platforms.rst]: Yes, thanks. I need to make several more corrections.

File: docs/reference/platforms.rst (hunk 2 above)
Comment (freakboy3742) [reply docs/reference/platforms.rst]: Both names are really valid; one is the module name, and one is the package name registered on PyPI - but the module name will also work as a package name.

**Pull request merged**